lib/virt_autotest/utils.pm

```diff
@@ -42,62 +42,89 @@ sub trim {
     return $text;
 }
 
-sub restart_libvirtd {
-    if (is_sle('<12')) {
-        assert_script_run('rclibvirtd restart', 180);
-    }
-    elsif (is_alp) {
-        my $_libvirtd_pid = script_output(q@ps -ef |grep [l]ibvirtd | gawk '{print $2;}'@);
-        my $_libvirtd_cmd = script_output("ps -o command $_libvirtd_pid | tail -1");
-        assert_script_run("kill -9 $_libvirtd_pid");
-        assert_script_run("$_libvirtd_cmd");
-    }
-    else {
-        systemctl("restart libvirtd", timeout => 180);
-    }
-    record_info("Libvirtd has been restarted!");
-    save_screenshot;
-}
-
 #return 1 if test is expected to run on XEN hypervisor
 sub is_xen_host {
     return get_var("XEN") || check_var("SYSTEM_ROLE", "xen") || check_var("HOST_HYPERVISOR", "xen") || check_var("REGRESSION", "xen-hypervisor");
 }
 
+# Usage: check_modular_libvirt_daemons([daemon1_name daemon2_name ...]). For example:
+# to specify daemons which will be checked: check_modular_libvirt_daemons(qemu storage ...)
+# to check all required modular daemons without any daemons passed
+sub check_modular_libvirt_daemons {
+    my @daemons = @_;
+
+    if (!@daemons) {
+        @daemons = qw(network nodedev nwfilter secret storage proxy log lock);
+        (is_xen_host) ? push @daemons, 'xen' : push @daemons, 'qemu';
+    }
+
+    foreach my $daemon (@daemons) {
+        systemctl("status virt${daemon}d.service");
+        if (($daemon == 'lock') || ($daemon == 'log')) {
+            systemctl("status virt${daemon}d\{,-admin\}.socket");
+        } else {
+            systemctl("status virt${daemon}d\{,-ro,-admin\}.socket");
+        }
+    }
+
+    record_info("Modular Libvirt daemons checked", join(' ', @daemons));
```

Comment:
Suggest to use msg "Modular libvirt daemons checked, all active for", then people without reading code would know all is good

Reply:
Sure, had updated msg info to make much more sense.